pkg/minikube/update/update.go

```diff
@@ -49,66 +50,100 @@ const (
 	githubRepo  = "minishift"
 )
 
-var (
-	lastUpdateCheckFilePath = constants.MakeMiniPath("last_update_check")
-)
-
-func MaybeUpdateFromGithub(output io.Writer) {
+// MaybeUpdateFromGithub checks if the local version is older than the latest version available from GitHub.
+// It returns an error if anything goes wrong during the update process
+func MaybeUpdateFromGithub(output io.Writer) error {
 	localVersion, err := version.GetSemverVersion()
 	if err != nil {
-		glog.Errorln(err)
-		return
+		return err
 	}
-	MaybeUpdate(output, githubOwner, githubRepo, githubRepo, lastUpdateCheckFilePath, localVersion)
+
+	latestVersion, err := getLatestVersionFromGitHub(githubOwner, githubRepo)
+	if err != nil {
+		return err
+	}
+
+	if isNewerVersion(localVersion, latestVersion) {
+		return MaybeUpdate(output, githubOwner, githubRepo, latestVersion)
+	}
+	return nil
 }
 
-func MaybeUpdate(output io.Writer, githubOwner, githubRepo, binaryName, lastUpdatePath string, localVersion semver.Version) {
+// isNewerVersion compares the two versions and returns a boolean
+func isNewerVersion(localVersion, latestVersion semver.Version) bool {
+	if localVersion.Compare(latestVersion) < 0 {
+		return true
+	} else {
+		return false
+	}
+}
+
+// MaybeUpdate handles the update process by calling functions that download, verify, extract and replace the binary.
+// It returns an error if any of these functions fails at any point.
+func MaybeUpdate(output io.Writer, githubOwner, githubRepo string, latestVersion semver.Version) error {
+
+	var extName string
+
+	if runtime.GOOS == "windows" {
+		extName = "zip"
+	} else {
+		extName = "tgz"
+	}
 
-	downloadBinary := binaryName + "-" + runtime.GOOS + "-" + runtime.GOARCH
-	updateLinkPrefix := "https://github.com/" + githubOwner + "/" + githubRepo + "/releases/tag/" + version.VersionPrefix
+	archiveName := fmt.Sprintf("minishift-%s-%s-%s.%s", latestVersion, runtime.GOOS, runtime.GOARCH, extName)
 	downloadLinkFormat := "https://github.com/" + githubOwner + "/" + githubRepo + "/releases/download/v%s/%s"
 
-	if !shouldCheckURLVersion(lastUpdatePath) {
-		return
+	fmt.Printf("Updating to version %s\n", latestVersion)
+
+	// downloadedAchive is a string that has the path to archive on the disk
+	downloadedArchive, err := downloadAndVerifyArchive(latestVersion, archiveName, downloadLinkFormat)
+	if err != nil {
+		return err
 	}
-	latestVersion, err := getLatestVersionFromGitHub(githubOwner, githubRepo)
+
+	dir := strings.Split(downloadedArchive, "minishift")[0]
+
+	defer os.RemoveAll(dir)
+
+	// if err := downloadAndVerifyChecksum(downloadedArchive, downloadLinkFormat, archiveName, latestVersion); err != nil {
+	// 	return err
+	// }
+
+	// Extract and replace the downloaded archive in place of current binary
+
+	binary, err := extractBinary(downloadedArchive)
 	if err != nil {
-		glog.Errorln(err)
-		return
+		return err
 	}
-	if localVersion.Compare(latestVersion) < 0 {
```

Comment:
This got completely dropped now. I think we this interactive step. We should get this back. I'd put that in the update command and expose from update.go

* CurrentVersion()
* LatestVersion()
* IsNewer(current, latest)
* Update (basically current MaybeUpdate)

This way you can implement the actual update logic in the update command. Determine whether there is a newer version, if not, display that Minishift is up to date. ATM, there is no feedback whatsoever, if your binary is up to date. 

If there is a newer binary, you ask for confirmation of the update. If update is confirmed, you make the call for updating the binary.